src/realm/alloc_slab.cpp

```diff
@@ -792,8 +794,8 @@ ref_type SlabAlloc::attach_file(const std::string& file_path, Config& cfg)
     // a later commit would have to do it. That would require coordination with
     // anybody concurrently joining the session, so it seems easier to do it at
     // session initialization, even if it means writing the database during open.
-    if (cfg.session_initiator && m_file_on_streaming_form) {
-        const Header& header = *reinterpret_cast<const Header*>(m_data);
+    const Header& header = *reinterpret_cast<const Header*>(m_data);
```

Comment:
If anything, wouldn't we need the `encryption_read_barrier()` before we use the header here?